HomeWork1/Maksim_Panibrat_HW1_Vector.hpp

```diff
@@ -0,0 +1,320 @@
+#pragma once
+
+// iostream used only in analize and prints, can be deleted when program will be complited
+#include <iostream>
+
+template <typename T>
+class Vector {
+public:
+	Vector() = default;
+	Vector(size_t new_size) : capacity{ new_size }, size{ capacity } {
+		data = alloc_memory(capacity);
+		new(data)T[capacity]{};
+	}
+	Vector(const size_t new_size, const T value) : capacity{ new_size }, size{ capacity } {
+		data = alloc_memory(capacity);
+		new(data)T[capacity];
```

Comment:
```suggestion
		data = alloc_memory(capacity);
		new(data)T[capacity];
```
у вас этот код повторяеться несколько раз, может стоит его вынести в отдельный метод.
Также можно использовать форвардинг конструкторов, у вас уже реализован функционал по выделению памяти в конструкторе `Vector(size_t new_size)`